modules/sdk-core/src/bitgo/utils/tss/baseTypes.ts

```diff
@@ -0,0 +1,132 @@
+  | 'pendingApproval'
+  | 'canceled'
+  | 'rejected'
+  | 'initialized'
+  | 'pendingDelivery'
+  | 'delivered'
+  | 'pendingUserSignature'
+  | 'signed';
+
+export type TransactionState =
+  | 'initialized'
+  | 'pendingSignature'
+  | 'signed'
+  | 'held'
+  | 'delivered'
+  | 'invalidSignature'
+  | 'rejected';
+
+/** Base TxRequest type that different scheme should override  */
+export type ITxRequest = {
+  txRequestId: string;
+  walletId: string;
+  walletType: WalletType;
+  version: number;
+  enterpriseId?: string;
+  state: TxRequestState;
+  date: string;
+  userId: string;
+  intent: unknown; // Should override with sig scheme specific intent
+  pendingApprovalId?: string;
+  policiesChecked: boolean;
+  signatureShares?: SignatureShareRecord[];
+  pendingTxHashes?: string[];
+  txHashes?: string[];
+  // Only available in 'lite' version
+  unsignedTxs: unknown[]; // Should override with blockchain / sig scheme specific unsigned tx
+  // Only available in 'full' version
+  transactions: {
+    state: TransactionState;
+    unsignedTx: unknown; // Should override with blockchain / sig specific unsigned tx
+    signatureShares: SignatureShareRecord[];
+  }[];
+  apiVersion?: TxRequestVersion;
+  latest: boolean;
+};
+
+export enum SignatureShareType {
+  USER = 'user',
+  BACKUP = 'backup',
+  BITGO = 'bitgo',
+}
+
+export interface SignatureShareRecord {
+  from: SignatureShareType;
+  to: SignatureShareType;
+  share: string;
+}
+
+/**
+ * Common Interface for implementing signature scheme specific
+ * util functions
+ */
+export interface ITssUtils<KeyShare = EDDSA.KeyShare, TxRequestType = EddsaTxRequest> {
+  createUserKeychain(
+    userGpgKey: SerializedKeyPair<string>,
+    userKeyShare: KeyShare,
+    backupKeyShare: KeyShare,
+    bitgoKeychain: Keychain,
+    passphrase: string,
+    originalPasscodeEncryptionCode: string,
+    recipientIndex?: number
+  ): Promise<Keychain>;
+  createBackupKeychain(
+    userGpgKey: SerializedKeyPair<string>,
+    userKeyShare: KeyShare,
+    backupKeyShare: KeyShare,
+    bitgoKeychain: Keychain,
+    passphrase: string
+  ): Promise<Keychain>;
+  createBitgoKeychain(
+    userGpgKey: SerializedKeyPair<string>,
+    userKeyShare: KeyShare,
+    backupKeyShare: KeyShare,
+    enterprise: string
+  ): Promise<Keychain>;
+  createKeychains(params: {
+    passphrase: string;
+    enterprise?: string;
+    originalPasscodeEncryptionCode?: string;
+  }): Promise<KeychainsTriplet>;
+  signTxRequest(params: {
+    txRequest: string | TxRequestType;
+    prv: string;
+    reqId: IRequestTracer;
+  }): Promise<TxRequestType>;
+  prebuildTxWithIntent(
+    params: PrebuildTransactionWithIntentOptions,
+    apiVersion?: TxRequestVersion,
+    preview?: boolean
+  ): Promise<TxRequestType>;
+  deleteSignatureShares(txRequestId: string): Promise<SignatureShareRecord[]>;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  sendTxRequest(txRequestId: string): Promise<any>;
```

Comment:
Linter started complaining for the use of `any`, would have preferred to use `unknown` but have a feeling that this change might break stuff which is already using this functions. Disabled check.